services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableQueryManagerImpl.java

```diff
@@ -169,7 +169,8 @@ SqlQuery queryPreflight(UserInfo user, Query query, Long maxBytesPerPage)
 		return new SqlQueryBuilder(model).tableSchema(columnModels).overrideOffset(query.getOffset())
 				.overrideLimit(query.getLimit()).maxBytesPerPage(maxBytesPerPage)
 				.includeEntityEtag(query.getIncludeEntityEtag()).selectedFacets(query.getSelectedFacets())
-				.sortList(query.getSort()).additionalFilters(query.getAdditionalFilters()).tableType(tableType).build();
+				.sortList(query.getSort()).additionalFilters(query.getAdditionalFilters()).tableType(tableType)
+				.userId(user.getId()).build();

```

Comment:
You might not want to make this parameter optional, and always include the id of the user in the constructor of the SqlQueryBuilder.